echo_server.py

```diff
@@ -96,4 +105,12 @@ def dir_list(uri):
     return body
 
 if __name__ == '__main__':
-    echo_server()
\ No newline at end of file
+    p = Process(target=echo_server, args=())
+    p.start()
+
+    j = Process(target=echo_client, args=(sys.argv[-1],))
+    j.start()
+    j.join
```

Comment:
It's not actually required that you run the client in a sub-process like this.  So long as the echo_server is running in one, the client can run in the parent process just fine.